ffi/cpp/include/decimal.hpp

```diff
@@ -68,6 +68,12 @@ class FixedDecimalFormat {
     return out;
   }
 
+  template<typename T> bool FormatToWriteable(const FixedDecimal& dec, T& writeable) const {
+    ICU4XWriteable writer = icu4x::internal::WriteableTrait<T>::Construct(writeable);
+    bool success = icu4x_fixed_decimal_format_write(this->inner.get(),
+                                                    dec.AsFFI(), &writer);
+    return success;
+  }
```

Comment:
Thought: Did you consider something more like

```cpp
bool FormatToWriteable(const FixedDecimal& dec, ICU4XWriteable& writeable) const { /* ... */ }

class ICU4XWriteable {
    ICU4XWriteable(std::string& s) { /* ... */ }
}
```

Then passing a `std::string` into `FormatToWriteable` will cause the `ICU4XWriteable` to be auto-constructed.

Reply:
Yes, I didn't want it to be dependent on other people being able to add constructors to `ICU4XWriteable`, since that's kind of weird privacy-wise. Specializing the `WriteableTrait` class is the standard pattern for this.